exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/internal/OtlpSpanExporter.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.otlp.internal;
+
+import io.opentelemetry.exporter.internal.marshal.Marshaler;
+import io.opentelemetry.exporter.internal.otlp.traces.LowAllocationTraceRequestMarshaler;
+import io.opentelemetry.exporter.internal.otlp.traces.TraceRequestMarshaler;
+import io.opentelemetry.exporter.otlp.stream.StreamExporter;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.common.export.MemoryMode;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import java.util.ArrayDeque;
+import java.util.Collection;
+import java.util.Deque;
+
+public class OtlpSpanExporter {
```

Comment:
Why build all this over again instead of extending [OtlpJsonLoggingSpanExporter](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/logging-otlp/src/main/java/io/opentelemetry/exporter/logging/otlp/OtlpJsonLoggingSpanExporter.java) with additional configuration options?

Reply:
I tried that first - it turned out that the code reuse is much easier to do with the otlp exporters, which have the same options, such as memory mode and metrics aggregation things. 

I didn't include the full code reuse, because that would make the pr bigger.

Reply:
What is different about this new exporter compared to `OtlpJsonLoggingSpanExporter`? Why can't we add a builder pattern to `OtlpJsonLoggingSpanExporter` to make the output stream configurable?

```
OtlpJsonLoggingSpanExporter.builder().setOutputStream(os).build();
```

Reply:
great point: the produced output is different.

- logging-json starts with `{ "record"` - https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/logging-otlp/src/test/java/io/opentelemetry/exporter/logging/otlp/OtlpJsonLoggingLogRecordExporterTest.java#L100
- otlp-stdout starts with `{"resourceLogs":` - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/file-exporter.md#examples

Reply:
great point: the produced output is different.

- logging-json starts with `{ "record"` - https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/logging-otlp/src/test/java/io/opentelemetry/exporter/logging/otlp/OtlpJsonLoggingLogRecordExporterTest.java#L100
- otlp-stdout starts with `{"resourceLogs":` - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/file-exporter.md#examples

Reply:
That seems like another config option:
```
OtlpJsonLoggingSpanExporter.builder()
  .setOutputStream(os)
  // Wrap Resource{Signal} with Export{Signal}ServiceRequest message
  .wrapWithExportRequestMessage() // ..or some other better method name
  .build();
```